test/cpp/c10d/CMakeLists.txt

```diff
@@ -39,6 +49,11 @@ if(USE_CUDA)
     c10d_add_test(
       ProcessGroupNCCLErrorsTest.cpp
       torch_cpu c10d_cuda_test gtest_main __caffe2_nccl)
+    if(INSTALL_TEST)
+      install(TARGETS ProcessGroupNCCLTest DESTINATION bin)
+      install(TARGETS ProcessGroupNCCLErrorsTest DESTINATION bin)
+      install(TARGETS c10d_cuda_test DESTINATION lib)
```

Comment:
curious why this one is not `bin` but `lib`

Reply:
I am a cmake beginner so not sure if this is a correct solution, but basically `ProcessGroupNCCLTest` needs `libc10d_cuda_test.so` library. 
* Currently, `ProcessGroupNCCLTest` is built and put at `build/bin`, `libc10d_cuda_test.so` is compiled and put at `build/lib`
* In this PR, I've added installing step, which copies `ProcessGroupNCCLTest` to `{env}/site-packages/torch/bin`, and `libc10d_cuda_test.so` to `{env}/site-packages/torch/lib` (again would have to reluctantly add sym link to make the tests pass)

Reply:
i guess the name is confusing it is not exactly a test but a test util library. 
but curious where was INSTALL_TEST defined in the first place? was it in another PR?

Reply:
`INSTALL_TEST` must be an option of the top-level CMakeLists.txt defined here https://github.com/pytorch/pytorch/blob/master/CMakeLists.txt#L174